anyway/parsers/registered.py

```diff
@@ -0,0 +1,156 @@
+    'בוקעאתה': 'בוקעאתא',
+    "ג'ש (גוש חלב)": "ג'ש )גוש חלב(",
+    'דבורייה': 'דבוריה',
+    'טובא-זנגרייה': 'טובא-זנגריה',
+    'יהוד': 'יהוד-מונוסון',
+    'יהוד-נווה אפרים': 'יהוד-מונוסון',
+    "כאוכב אבו אל- היג'א": "כאוכב אבו אל-היג'א",
+    "כעביה-טבאש- חג'אג'רה": "כעביה-טבאש-חג'אג'רה",
+    'מודיעין-מכבים- רעות' : 'מודיעין-מכבים-רעות',
+    'נהרייה' : 'נהריה',
+    "פקיעין (בוקייעה)" : "פקיעין )בוקייעה(",
+    'פרדסייה' : 'פרדסיה',
+    'קציר-חריש' : 'קציר',
+    'שבלי-אום אל-גנם' : 'שבלי - אום אל-גנם',
+}
+
+
+class DatastoreImporter(object):
+    _header_size = 12
+    _in_encode = "utf-8"
+    _report_year = 0
+    _population_year = 0
+
+    def import_file(self, inputfile):
+        total = 0
+        elements = os.path.basename(inputfile).split('_')
+        self._report_year = self.as_int(elements[0])
+        with io.open(inputfile, 'r', encoding=self._in_encode) as csvfile:
+            csvreader = csv.reader(csvfile, delimiter=',', quotechar='"')
+            row_count = 1
+            inserts = []
+            for row in csvreader:
+                if row_count > self._header_size:
+                    if self.is_process_row(row):
+                        total += 1
+                        inserts.append(self.row_parse(row))
+                else:
+                    self.header_row(row)
+                row_count += 1
+
+            db.session.bulk_insert_mappings(RegisteredVehicle, inserts)
+        return total
+
+    @staticmethod
+    def is_process_row(row):
+        if row[0].strip() == '' or row[1].strip() == '':
+            return False
+        return True
+
+    def row_parse(self, row):
+        name = row[12].strip().encode('utf-8')
+        name = re.sub(' +', ' ', name).replace('קריית', 'קרית').replace("\n", '')
+        search_name = name
+        if name in _manual_assing_on_city_name:
+            search_name = _manual_assing_on_city_name[name]
+
+        return {
+            'year': self._report_year,
+            'name': name,
+            'name_eng': row[0].strip(),
+            'search_name' : search_name,
+            'motorcycle': self.as_int(row[1]),
+            'special': self.as_int(row[2]),
+            'taxi': self.as_int(row[3]),
+            'bus': self.as_int(row[4]),
+            'minibus': self.as_int(row[5]),
+            'truck_over3500': self.as_int(row[6]),
+            'truck_upto3500': self.as_int(row[7]),
+            'private': self.as_int(row[9]),
+            'total': self.as_int(row[10]),
+            'population': self.as_int(row[11]),
+            'population_year': self._population_year
+        }
+
+    @staticmethod
+    def as_int(value):
+        value = value.strip().replace(',', '')
+        try:
+            return int(value)
+        except ValueError:
+            return 0
+
+    def header_row(self, row):
+        if row[1].strip() == "cycle":
+            self._population_year = self.as_int(row[11])
+
+
+def main(specific_folder, delete_all, path):
```

Comment:
This part should be shared with `cbs` (maybe move to `utilities`) if not moved to the same file.

Reply:
yes & no.
the code not quite the same:
the 'specific_folder' is same but 'delete_all' is not & what comes next.

The question here is in my current scope (first issue in project) is will be wise for me to refactor part of cbs... 
i have no problem to do it, but I am new to python and my be my way of OOP is not fit to standards...

Reply:
The common part should be refactored to a single function. If there are differences, they can be controlled by parameters. Don't worry about this being your first issue, as we are doing this code review to check if the changes make sense, and we also have tests.

Reply:
I refactor my code to new common solution, if it ok, i will refactor cbs for it

Reply:
ok, refactoring - if it ok, i will change cbs too